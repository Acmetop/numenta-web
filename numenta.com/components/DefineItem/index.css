/*Numenta company website source code. Copyright Â© 2016 Numenta.
Full details in LICENSE.txt, or contact us at <http://numenta.com>.
This program is free software: you can redistribute it and/or modify it under
the terms of the GNU Affero General Public License as published by the Free
Software Foundation, either version 3 of the License, or (at your option) any
later version. This program is distributed in the hope that it will be
useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
General Public License for more details. You should have received a copy of
the GNU Affero General Public License along with this program. If not, see
<https://www.gnu.org/licenses/agpl.html>.*/

.horizontal {
  composes: fl from 'tachyons';
  composes: h-100 from 'tachyons';
  composes: ma0 from 'tachyons';
  padding-bottom: 0.2rem;
  padding-top: 0.4rem;
  width: 65%;
}

.vertical {
  composes: mb3 from 'tachyons';
  composes: ml0 from 'tachyons';
  composes: pl0 from 'tachyons';
}


.defineItem {
  word-wrap: break-word;
}
