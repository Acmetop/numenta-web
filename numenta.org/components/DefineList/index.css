/**
 * Numenta.org HTM Community website source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2016 Numenta <http://numenta.com>
 */

.horizontal {
  composes: cf from 'tachyons';
  composes: lh-title from 'tachyons';
}

.vertical {
  composes: lh-copy from 'tachyons';
  composes: mb3 from 'tachyons';
  composes: measure from 'tachyons';
  composes: measure-wide-l from 'tachyons';
}


.defineList {
  composes: ma0 from 'tachyons';
  composes: pa0 from 'tachyons';
}
