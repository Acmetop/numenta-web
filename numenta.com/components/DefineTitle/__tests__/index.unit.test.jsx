import React from 'react'
import renderer from 'react-test-renderer'

import DefineItem from '../../DefineItem'
import DefineList from '../../DefineList'
import DefineTitle from '../../DefineTitle'


describe('<DefineTitle /> React component unit test suite', () => {

  it('Renders correctly', () => {
    const component = renderer.create(
      <DefineList>
        <DefineTitle>
          Title 1
        </DefineTitle>
        <DefineItem>
          Item 1
        </DefineItem>
        <DefineTitle>
          Title 2
        </DefineTitle>
        <DefineItem>
          Item 2
        </DefineItem>
      </DefineList>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
