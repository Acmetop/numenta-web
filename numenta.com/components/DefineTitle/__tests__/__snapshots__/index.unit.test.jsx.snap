exports[`<DefineTitle /> React component unit test suite Renders correctly 1`] = `
<dt
  className="defineTitle" />
`;
